File: 21.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if not l1: return l2
        if not l2: return l1
        l3 = h3 = ListNode(0)
        while l1 and l2:
            if l1.val > l2.val:
                l3.next = ListNode(l2.val)
                l3, l2 = l3.next, l2.next
            else:
                l3.next = ListNode(l1.val)
                l3, l1 = l3.next, l1.next
        l3.next = l1 or l2
        return h3.next

File: 21.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* list3 = new ListNode(0);
        ListNode* list4 = list3;
        while(list1 and list2){
            if(list1->val > list2->val){
                list3->next = list2;
                list2 = list2->next;
                list3 = list3->next;
            }else{
                list3->next = list1;
                list1 = list1->next;
                list3 = list3->next;
            }
        }
        if (list1){
            list3->next = list1;
        }
        if (list2){
            list3->next = list2;
        }
        return list4->next;
    }
};
